tests/tests.rs

```diff
@@ -137,32 +135,6 @@ fn basic_memory() -> Result<()> {
     )
 }
 
-#[test]
-fn multi_memory() -> Result<()> {
-    run_wat(
-        &[],
-        42,
-        r#"
-(module
-  (memory $m1 1)
-  (memory $m2 1)
-  (func (export "wizer.initialize")
-    i32.const 0
-    i32.const 41
-    i32.store (memory $m1) offset=1337
-    i32.const 0
-    i32.const 1
-    i32.store (memory $m2) offset=1337)
-  (func (export "run") (result i32)
-    i32.const 0
-    i32.load (memory $m1) offset=1337
-    i32.const 0
-    i32.load (memory $m2) offset=1337
-    i32.add))
-"#,
-    )
-}
```

Comment:
This test shouldn't be deleted, we should still support multi-memory Wasm.

Reply:
I've re-added it now, the test fails but I unfortunately don't know how to make it pass.

```
failures:

---- multi_memory stdout ----
Error: unknown operator or unexpected token
     --> <anon>:8:16
      |
    8 |     i32.store (memory $m1) offset=1337
      |                ^


failures:
    multi_memory
```

Reply:
It looks like the WAT syntax for multi-memory has changed so that should just be

```
i32.store $m1 offset=1337
```

now.

Reply:
Thanks, I made that change to both the `store` calls and both the `load` calls however I believe it fails to generate valid wasm. The error that is generated is
>Error: unknown memory 1337 (at offset 75)

This error comes from `store` call for the `$m2` memory.

If I change the calls to use different offsets from each other then the test passes but I'm not sure if that is just avoiding a potential bug that I've introduced within this PR by updating wasmtime.

If useful, this is the wat which fails to validate:
```wat
(module
  (memory $m1 1)
  (memory $m2 1)
  (func (export "wizer.initialize")
    i32.const 0
    i32.const 41
    i32.store $m1 offset=1337
    i32.const 0
    i32.const 1
    i32.store $m2 offset=1337)
  (func (export "run") (result i32)
    i32.const 0
    i32.load $m1 offset=1337
    i32.const 0
    i32.load $m2 offset=1337
    i32.add))
```

And this is the wat which passes the test:
```wat
(module
  (memory $m1 1)
  (memory $m2 1)
  (func (export "wizer.initialize")
    i32.const 0
    i32.const 41
    i32.store $m1 offset=1337
    i32.const 0
    i32.const 1
    i32.store $m2 offset=1)
  (func (export "run") (result i32)
    i32.const 0
    i32.load $m1 offset=1337
    i32.const 0
    i32.load $m2 offset=1
    i32.add))
```

Reply:
Huh. That looks correct by my reading of [the parsing code](https://github.com/bytecodealliance/wasm-tools/blob/469a9107f447e3024b2d4e5ed047a570732792e3/crates/wast/src/core/expr.rs#L1301-L1358) but it looks like it is trying to treat the offset (static index into the memory) as a memory index (thing that decides which memory we are accessing).

Reply:
Actually it looks like the parser only supports memory indices, not IDs. So this should work:

```wat
i32.load 0 offset=1337 ;; $m1 is index 0
i32.load 1 offset=1337 ;; $m2 is index 1
```

Reply:
Actually IDs should work too.

In fact, if I copy/paste the failing example and run our validator on it, it succeeds:

```
$ cat ~/scratch/multi-memory.wat 
(module
 (memory $m1 1)
 (memory $m2 1)
 (func (export "wizer.initialize")
       i32.const 0
       i32.const 41
       i32.store $m1 offset=1337
       i32.const 0
       i32.const 1
       i32.store $m2 offset=1337)
 (func (export "run") (result i32)
       i32.const 0
       i32.load $m1 offset=1337
       i32.const 0
       i32.load $m2 offset=1337
       i32.add))
$ wasm-tools validate --features multi-memory ~/scratch/multi-memory.wat 
$ echo $?
0
```

Reply:
Are you sure that you tested that exact version of the WAT and not accidentally some in between version while editing and poking at the test case?

Reply:
Hmm maybe wasm-tools have changed since wasmtime 3.0 and 4.0? If you rebase your 4.0 work on the latest version of this branch, do you get the same failure?

Reply:
I rebased on wasmtime 4 branch and the same test failure happens.

I went to see what the prewizened wasm looked like after parsing the wat, and this is what it is:
```
 === PreWizened Wasm ==========================================================
    (module
      (type (;0;) (func))
      (type (;1;) (func (result i32)))
      (func (;0;) (type 0)
        i32.const 0
        i32.const 41
        i32.store offset=1337
        i32.const 0
        i32.const 1
        i32.store (memory 1337) offset=1)
      (func (;1;) (type 1) (result i32)
        i32.const 0
        i32.load offset=1337
        i32.const 0
        i32.load (memory 1337) offset=1
        i32.add)
      (memory $m1 1)
      (memory $m2 1)
      (export "wizer.initialize" (func 0))
      (export "run" (func 1)))
    ===========================================================================
```